sympy/integrals/intpoly.py

```diff
@@ -0,0 +1,806 @@
+
+    if isinstance(expr, Expr) and not expr.is_number:
+        if expr.is_Add:
+            for monomial in expr.args:
+                degree = 0
+                if monomial.is_Pow:
+                    degree += monomial.args[1]
+                else:
+                    term_type = len(monomial.args)
+                    if term_type == 0:
+                        if monomial.is_Symbol:
+                            degree += 1
+                    for univariate in monomial.args:
+                        term_type = len(univariate.args)
+                        if term_type == 0 and univariate.is_Symbol:
+                            degree += 1
+                        elif term_type == 2:
+                            degree += univariate.args[1]
+                if separate:
+                    monoms.append(monomial)
+                else:
+                    if degree in poly_dict:
+                        poly_dict[degree] += monomial
+                    else:
+                        poly_dict[degree] = monomial
+        elif expr.is_Mul:
+            degree = 0
+            for term in expr.args:
+                term_type = len(term.args)
+                if term_type == 0 and term.is_Symbol:
+                    degree += 1
+                elif term_type == 2:
+                    degree += term.args[1]
+            if separate:
+                monoms.append(expr)
+            else:
+                poly_dict[degree] = expr
+        elif expr.is_Pow:
+            if separate:
+                monoms.append(expr)
+            else:
+                poly_dict[expr.args[1]] = expr
+        else:
+            if separate:
+                monoms.append(expr)
+            else:
+                poly_dict[1] = expr
+    else:
+        if separate:
+            monoms.append(expr)
+        else:
+            poly_dict[0] = expr
+
+    if separate:
+        return monoms
+    return poly_dict
+
+
+def clockwise_sort(poly):
+    """Returns the same polygon with points sorted in clockwise order.
+
+    Note that it's necessary for input points to be sorted in some order
+    (clockwise or anti-clockwise) for the algorithm to work. As a convention
+    algorithm has been implemented keeping clockwise orientation in mind.
+
+    Parameters
+    ==========
+    poly: 2-Polytope
+
+    Examples
+    ========
+    >>> from sympy.integrals.intpoly import clockwise_sort
+    >>> from sympy.geometry.point import Point
+    >>> from sympy.geometry.polygon import Polygon
+    >>> clockwise_sort(Polygon(Point(0, 0), Point(1, 0), Point(1, 1)))
+    Triangle(Point2D(1, 1), Point2D(1, 0), Point2D(0, 0))
+
+    """
+    n = len(poly.vertices)
+    vertices = list(poly.vertices)
+    center = Point(sum(map(lambda vertex: vertex.x, poly.vertices)) / n,
+                   sum(map(lambda vertex: vertex.y, poly.vertices)) / n)
+
+    def compareTo(a, b):
+        if a.x - center.x >= 0 and b.x - center.x < 0:
+            return -1
```

Comment:
Unless there is some reason, I think it is better if sympified objects are returned, at other places as well in the code.